### ISSUE TITLE
Cache building compile error

### ISSUE THREAD

**Issue by johngrogg** [cloudfoundry/nodejs-buildpack#83]
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?
{
   "name": "",
   "build": "",
   "support": "https://support.run.pivotal.io",
   "version": 0,
   "description": "Cloud Foundry sponsored by Pivotal",
   "authorization_endpoint": "https://login.run.pivotal.io",
   "token_endpoint": "https://uaa.run.pivotal.io",
   "min_cli_version": "6.22.0",
   "min_recommended_cli_version": "latest",
   "api_version": "2.80.0",
   "app_ssh_endpoint": "ssh.run.pivotal.io:2222",
   "app_ssh_host_key_fingerprint": "e7:13:4e:32:ee:39:62:df:54:41:d7:f7:8b:b2:a7:6b",
   "app_ssh_oauth_client": "ssh-proxy",
   "routing_endpoint": "https://api.run.pivotal.io/routing",
   "logging_endpoint": "wss://loggregator.run.pivotal.io:443",
   "doppler_logging_endpoint": "wss://doppler.run.pivotal.io:443",
   "user": "f91c356d-779f-4bbe-8dcd-379054b57ea8"
}
cf version 6.25.0+787326d95.2017-02-28

What version of the buildpack you are using?
1.5.33

If you were attempting to accomplish a task, what was it you were attempting to do?
Push a nodejs app to Pivotal Web Services

What did you expect to happen?
The push to execute successfully like it did on buildpack v1.5.32.

What was the actual behavior?
Saw the following error:
```
-----> Caching build
error: Invalid character
.cacheDirectories | .[]?
                       ^
1 compile error
error: Invalid character
.cache_directories | .[]?
                        ^
1 compile error
```
Seemingly as a result of this error, cf run-task executions cannot find the node application.

Please confirm where necessary:
* [x] I have included a log output
* [x] My log includes an error message
* [ ] I have included steps for reproduction

**Comment by johngrogg** [comment]
A bit of follow up detail. Pushing our app with v1.5.32 of the nodejs buildpack results in a successful deploy with no errors and the cf run-task executions having access to the node executable.

**Comment by sclevine** [comment]
Thanks for the report, we'll take a look. Do you have a sample app that consistently demonstrates the behavior?

**Comment by johngrogg** [comment]
Not that is public, but here's a package.json that should hopefully exhibit the issue:

```
{
  "name": "na",
  "version": "1.0.0",
  "description": "na",
  "main": "app.js",
  "scripts": {
    "start": "node app.js"
  },
  "author": "",
  "license": "ISC",
  "engines": {
    "node": "6.10.X"
  },
  "dependencies": {
    "async": "^2.1.4",
    "bluebird": "^3.5.0",
    "body-parser": "^1.15.2",
    "cfenv": "^1.0.3",
    "colors": "^1.1.2",
    "command-line-args": "^4.0.2",
    "command-line-usage": "^4.0.0",
    "cookie-parser": "^1.4.3",
    "cookie-session": "^2.0.0-beta.1",
    "express": "^4.14.0",
    "get-port": "^3.1.0",
    "glob": "^7.1.1",
    "method-override": "^2.3.7",
    "moment": "^2.17.1",
    "moment-timezone": "^0.5.12",
    "morgan": "^1.7.0",
    "mysql": "^2.12.0",
    "passport": "^0.3.2",
    "passport-local": "^1.0.0",
    "request": "^2.79.0",
    "sequelize": "^3.29.0",
    "tunnel-ssh": "^4.1.2",
    "umzug": "^1.11.0"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-as-promised": "^6.0.0",
    "chai-string": "^1.3.0",
    "grunt": "^1.0.1",
    "grunt-concurrent": "^2.3.1",
    "grunt-contrib-jshint": "^1.1.0",
    "grunt-contrib-watch": "^1.0.0",
    "grunt-express-server": "^0.5.3",
    "grunt-mocha-test": "^0.13.2",
    "grunt-notify": "^0.4.5",
    "jshint-stylish": "^2.2.1",
    "mocha": "^3.2.0",
    "mocha-junit-reporter": "https://registry.npmjs.org/mocha-junit-reporter/-/mocha-junit-reporter-1.13.0.tgz",
    "sinon": "^1.17.6",
    "sinon-as-promised": "^4.0.2",
    "sinon-chai": "^2.8.0"
  }
}
```

**Comment by sesmith177** [comment]
I think the 'jq' error is a bit of a red herring as it won't (on its own) cause the build to fail, e.g

```
-----> Caching build
error: Invalid character
.cacheDirectories | .[]?
                       ^
1 compile error
error: Invalid character
.cache_directories | .[]?
                        ^
1 compile error
       Clearing previous node cache
       Saving 3 cacheDirectories (default):
       - .npm
       - .cache/yarn (nothing to cache)
       - bower_components (nothing to cache)
-----> Build succeeded!
Exit status 0
Staging complete
Uploading droplet, build artifacts cache...
Uploading build artifacts cache...
Uploading droplet...
Uploaded build artifacts cache (511.8K)
Uploaded droplet (17.6M)
Uploading complete
Destroying container
Successfully destroyed container

1 of 1 instances running

App started
```

Can you show us the output of `cf logs --recent` after the app fails to stage?

**Comment by johngrogg** [comment]
Sorry for the confusion. The app doesn't fail to stage. The blocking problem is that cf run-task can't find the node executable. I just assumed that the compile error in the build was the cause of that also.

**Comment by sclevine** [comment]
Thanks for the clarification -- if staging isn't broken, then we'll address this next week.

**Comment by RadhamaniK** [comment]
Hi Team, 

We are facing problem in accessing node environment variables. Because of that we are facing issues in production. It would be better if you can address this as soon as possible. Thanks

**Comment by sclevine** [comment]
It seems that the `LANG` env var is set to `en_US.UTF-8` by Cloud Controller for app launch and `cf ssh`, but not for tasks. This causes `.profile.d` scripts to be sourced in the wrong order. Explicitly setting that env var on your app can serve as a workaround until the next release, or until we have a fix in Cloud Controller. Alternatively, you can use an older buildpack version.

**Comment by RadhamaniK** [comment]
Thanks for the information.. Could you please let me know which buildpack version is stable to clone ?

**Comment by harini2016** [comment]
Hello All,

I have a same the problem when trying to build our node.js app. 

-----> Caching build
error: Invalid character
.cacheDirectories | .[]?
                       ^
1 compile error
error: Invalid character
.cache_directories | .[]?
                        ^
1 compile error
       Clearing previous node cache
       Saving 3 cacheDirectories (default):
       - .npm
       - .cache/yarn (nothing to cache)
       - bower_components (nothing to cache)

I ran the cf curl /v2/info and cf version. Here is the output:

{
   "name": "",
   "build": "",
   "support": "http://support.cloudfoundry.com",
   "version": 0,
   "description": "Cloud Foundry GE Open Sandbox",
   "authorization_endpoint": "https://login.system.aws-usw02-pr.ice.predix.io",
   "token_endpoint": "https://uaa.system.aws-usw02-pr.ice.predix.io",
   "min_cli_version": null,
   "min_recommended_cli_version": null,
   "api_version": "2.62.0",
   "app_ssh_endpoint": "ssh.system.aws-usw02-pr.ice.predix.io:2222",
   "app_ssh_host_key_fingerprint": "a6:d1:08:0b:b0:cb:9b:5f:c4:ba:44:2a:97:26:19:8a",
   "app_ssh_oauth_client": "ssh-proxy",
   "logging_endpoint": "wss://loggregator.system.aws-usw02-pr.ice.predix.io:443",
   "doppler_logging_endpoint": "wss://doppler.system.aws-usw02-pr.ice.predix.io:443",
   "user": "668ec2d9-4e88-4c0f-9191-4f7fb83182df"
}

cf version 6.18.0+b22884b-2016-05-10

Is there any temporary fix for this problem? Due to this all our apps went down!

**Comment by sclevine** [comment]
Confirmed that the task issue is a bug in CC, when the new consumed bridge is used. The workaround is to set `LANG` to `en_US.UTF-8` on your app until there is a CC fix available.

@harini2016 are those errors during a failed `cf push` that succeeds with an older version of the buildpack?

**Comment by harini2016** [comment]
I am not sure about if the team has updated out buildpack. I started seeing this recently from Apr 27. Yes it is during the cf push.

How do i set that LANG. What is the cf command we should use ?

**Comment by sclevine** [comment]
If you're experiencing the failure during `cf push`, then it's not related to the `LANG` issue. Can you provide staging logs from an application that's failing?

**Comment by sclevine** [comment]
@RadhamaniK [v1.5.32](https://github.com/cloudfoundry/nodejs-buildpack/releases/tag/v1.5.32) doesn't depend on the value of `LANG` during tasks. We will likely cut a v1.5.34 that works around the unexpected behavior from CC.

**Comment by harini2016** [comment]
-------> Buildpack version 1.5.33
-----> Installing binaries
       engines.node (package.json):  ^6.3.1
       engines.npm (package.json):   ^3.10.6
       Downloading and installing node 6.10.3...
       Downloaded [https://buildpacks.cloudfoundry.org/dependencies/node/node-6.10.3-linux-x64-34d4554a.tgz]
       /tmp/staged/app/.cloudfoundry/0/bin /tmp/staged/app
       /tmp/staged/app
       Downloading and installing npm ^3.10.6 (replacing version 3.10.10)...
       Downloading and installing yarn (0.23.4)...
       Downloaded [https://buildpacks.cloudfoundry.org/dependencies/yarn/yarn-v0.23.4-e6837e4c.tar.gz]
       /tmp/staged/app/.cloudfoundry/0/bin /tmp/staged/app
       /tmp/staged/app
       Installed yarn 0.23.4
-----> Writing config
       PRO TIP: It is recommended to vendor the application's Node.js dependencies
                See http://docs.cloudfoundry.org/buildpacks/node/index.html#vendoring for more information
-----> Creating runtime environment
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
       NODE_VERBOSE=false
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       NODE_HOME=/tmp/staged/app/.cloudfoundry/0/node
-----> Restoring cache
-----> Building dependencies
       Skipping cache restore (new runtime signature)
       Installing node modules (package.json)
       flydubai-ui-app-hub@1.0.0 /tmp/staged/app
       ├─┬ compression@1.6.2
       │ ├─┬ accepts@1.3.3
       │ │ └── negotiator@0.6.1
       │ ├── bytes@2.3.0
       │ ├─┬ compressible@2.0.10
       │ │ └── mime-db@1.27.0
       │ ├── debug@2.2.0
       │ └── vary@1.1.1
       ├── connect-redis@3.3.0
       ├─┬ cookie-parser@1.4.3
       │ ├── cookie@0.3.1
       │ └── cookie-signature@1.0.6
       ├─┬ debug@2.6.6
       │ └── ms@0.7.3
       │ ├── array-flatten@1.1.1
       │ ├── content-disposition@0.5.2
       │ ├── on-headers@1.0.1
       ├─┬ express@4.14.1
       │ ├── content-type@1.0.2
       │ ├── debug@2.2.0
       │ ├── depd@1.1.0
       │ ├── encodeurl@1.0.1
       │ ├── escape-html@1.0.3
       │ ├── etag@1.7.0
       │ ├─┬ finalhandler@0.5.1
       │ │ ├── debug@2.2.0
       │ │ ├── statuses@1.3.1
       │ │ └── unpipe@1.0.0
       │ ├── fresh@0.3.0
       │ ├── merge-descriptors@1.0.1
       │ ├── methods@1.1.2
       │ ├─┬ on-finished@2.3.0
       │ │ └── ee-first@1.1.1
       │ ├── parseurl@1.3.1
       │ ├── path-to-regexp@0.1.7
       │ ├─┬ proxy-addr@1.1.4
       │ │ ├── forwarded@0.1.0
       │ │ └── ipaddr.js@1.3.0
       │ ├── qs@6.2.0
       │ ├── range-parser@1.2.0
       │ ├─┬ send@0.14.2
       │ │ ├─┬ debug@2.2.0
       │ │ │ └── ms@0.7.1
       │ │ ├── destroy@1.0.4
       │ │ ├─┬ http-errors@1.5.1
       │ │ │ └── setprototypeof@1.0.2
       │ │ ├── mime@1.3.4
       │ │ └── ms@0.7.2
       │ ├── serve-static@1.11.2
       │ ├─┬ type-is@1.6.15
       │ │ └── media-typer@0.3.0
       │ └── utils-merge@1.0.0
       ├── express-bearer-token@2.1.0
       ├─┬ express-handlebars@3.0.0
       │ ├─┬ glob@6.0.4
       │ │ ├─┬ inflight@1.0.6
       │ │ │ └── wrappy@1.0.2
       │ │ ├── inherits@2.0.3
       │ │ ├─┬ minimatch@3.0.3
       │ │ │ └─┬ brace-expansion@1.1.7
       │ │ │   ├── balanced-match@0.4.2
       │ │ │   └── concat-map@0.0.1
       │ │ ├── once@1.4.0
       │ │ └── path-is-absolute@1.0.1
       │ ├── graceful-fs@4.1.11
       │ ├─┬ handlebars@4.0.8
       │ │ ├─┬ optimist@0.6.1
       │ │ │ ├── minimist@0.0.10
       │ │ │ └── wordwrap@0.0.3
       │ │ ├─┬ source-map@0.4.4
       │ │ │ └── amdefine@1.0.1
       │ │ └─┬ uglify-js@2.8.22
       │ │   ├── source-map@0.5.6
       │ │   ├── uglify-to-browserify@1.0.2
       │ │   └─┬ yargs@3.10.0
       │ │     ├── camelcase@1.2.1
       │ │     ├─┬ cliui@2.1.0
       │ │     │ ├─┬ center-align@0.1.3
       │ │     │ │ ├─┬ align-text@0.1.4
       │ │     │ │ │ └── longest@1.0.1
       │ │     │ │ └── lazy-cache@1.0.4
       │ │     │ ├── right-align@0.1.3
       │ │     │ └── wordwrap@0.0.2
       │ │     ├── decamelize@1.2.0
       │ │     └── window-size@0.1.0
       │ ├─┬ object.assign@4.0.4
       │ │ ├─┬ define-properties@1.1.2
       │ │ │ └── foreach@2.0.5
       │ │ ├── function-bind@1.1.0
       │ │ └── object-keys@1.0.11
       │ └─┬ promise@7.1.1
       │   └── asap@2.0.5
       ├─┬ express-http-proxy@0.10.1
       │ ├── es6-promise@3.3.1
       │ └─┬ raw-body@2.2.0
       │   ├── bytes@2.4.0
       │   └── iconv-lite@0.4.15
       ├─┬ express-request-proxy@2.0.0
       │ ├── async@1.5.2
       │ ├─┬ body-parser@1.17.1
       │ │ ├── bytes@2.4.0
       │ │ ├─┬ debug@2.6.1
       │ │ │ └── ms@0.7.2
       │ │ ├─┬ http-errors@1.6.1
       │ │ │ └── setprototypeof@1.0.3
       │ │ └── qs@6.4.0
       │ ├─┬ camel-case@1.2.2
       │ │ ├─┬ sentence-case@1.1.3
       │ │ │ └── lower-case@1.1.4
       │ │ └── upper-case@1.1.3
       │ ├─┬ lru-cache@4.0.2
       │ │ ├── pseudomap@1.0.2
       │ │ └── yallist@2.1.2
       │ ├─┬ path-to-regexp@1.7.0
       │ │ └── isarray@0.0.1
       │ ├─┬ simple-errors@1.0.1
       │ │ └─┬ errno@0.1.4
       │ │   └── prr@0.0.0
       │ ├─┬ through2@2.0.3
       │ │ └─┬ readable-stream@2.2.9
       │ │   ├── buffer-shims@1.0.0
       │ │   ├── isarray@1.0.0
       │ │   ├── process-nextick-args@1.0.7
       │ │   ├── string_decoder@1.0.0
       │ │   └── util-deprecate@1.0.2
       │ └── url-join@0.0.1
       ├─┬ express-session@1.15.2
       │ ├── crc@3.4.4
       │ ├─┬ debug@2.6.3
       │ │ └── ms@0.7.2
       │ └─┬ uid-safe@2.1.4
       │   └── random-bytes@1.0.0
       ├─┬ helmet@1.3.0
       │ ├─┬ connect@3.4.1
       │ │ ├── debug@2.2.0
       │ │ └── finalhandler@0.4.1
       │ ├── dns-prefetch-control@0.1.0
       │ ├── dont-sniff-mimetype@1.0.0
       │ ├─┬ frameguard@1.1.0
       │ │ └── lodash.isstring@4.0.1
       │ ├─┬ helmet-csp@1.1.0
       │ │ ├── camelize@1.0.0
       │ │ ├─┬ content-security-policy-builder@1.0.0
       │ │ │ └── dashify@0.2.2
       │ │ ├─┬ lodash.assign@4.0.4
       │ │ │ ├── lodash.keys@4.2.0
       │ │ │ └── lodash.rest@4.0.5
       │ │ ├── lodash.isfunction@3.0.8
       │ │ ├─┬ lodash.reduce@4.2.0
       │ │ │ ├── lodash._baseeach@4.1.3
       │ │ │ ├─┬ lodash._baseiteratee@4.7.0
       │ │ │ │ └─┬ lodash._stringtopath@4.8.0
       │ │ │ │   └── lodash._basetostring@4.12.0
       │ │ │ └── lodash._basereduce@3.0.2
       │ │ ├── lodash.some@4.2.0
       │ │ └── platform@1.3.1
       │ ├── hide-powered-by@1.0.0
       │ ├── hpkp@1.1.0
       │ ├─┬ hsts@1.0.0
       │ │ └── core-util-is@1.0.2
       │ ├── ienoopen@1.0.0
       │ ├── nocache@1.0.0
       │ └── x-xss-protection@1.0.0
       ├─┬ http-proxy-middleware@0.15.2
       │ ├─┬ http-proxy@1.16.2
       │ │ ├── eventemitter3@1.2.0
       │ │ └── requires-port@1.0.0
       │ ├─┬ is-glob@2.0.1
       │ │ └── is-extglob@1.0.0
       │ └─┬ micromatch@2.3.11
       │   ├─┬ arr-diff@2.0.0
       │   │ └── arr-flatten@1.0.3
       │   ├── array-unique@0.2.1
       │   ├─┬ braces@1.8.5
       │   │ ├─┬ expand-range@1.8.2
       │   │ │ └─┬ fill-range@2.2.3
       │   │ │   ├── is-number@2.1.0
       │   │ │   ├─┬ isobject@2.1.0
       │   │ │   │ └── isarray@1.0.0
       │   │ │   ├── randomatic@1.1.6
       │   │ │   └── repeat-string@1.6.1
       │   │ ├── preserve@0.2.0
       │   │ └── repeat-element@1.1.2
       │   ├─┬ expand-brackets@0.1.5
       │   │ └── is-posix-bracket@0.1.1
       │   ├── extglob@0.3.2
       │   ├── filename-regex@2.0.1
       │   ├─┬ kind-of@3.2.0
       │   │ └── is-buffer@1.1.5
       │   ├─┬ normalize-path@2.1.1
       │   ├─┬ object.omit@2.0.1
       │   │ ├─┬ for-own@0.1.5
       │   │ │ └── for-in@1.0.2
       │   │ └── is-extendable@0.1.1
       │   ├─┬ parse-glob@3.0.4
       │   │ ├─┬ glob-base@0.3.0
       │   │ │ └── glob-parent@2.0.0
       │   │ └── is-dotfile@1.0.2
       │     ├── is-equal-shallow@0.1.3
       │     └── is-primitive@2.0.0
       │   │ └── remove-trailing-separator@1.0.1
       │   └─┬ regex-cache@0.4.3
       ├── http-status-codes@1.1.6
       ├── i18next@3.5.2
       ├─┬ i18next-express-middleware@1.0.5
       │ └─┬ cookies@0.6.1
       │   └── keygrip@1.0.1
       ├─┬ i18next-node-fs-backend@0.1.3
       │ ├─┬ js-yaml@3.5.4
       │ │ ├─┬ argparse@1.0.9
       │ │ └── esprima@2.7.3
       │ └── json5@0.5.0
       ├─┬ jsonwebtoken@5.7.0
       │ ├─┬ jws@3.1.4
       │ │ ├── base64url@2.0.0
       │ │ └─┬ jwa@1.1.5
       │ │   ├── buffer-equal-constant-time@1.0.1
       │ │   └── ecdsa-sig-formatter@1.0.9
       │ ├── ms@0.7.1
       ├── lodash@4.17.4
       │ ├── basic-auth@1.1.0
       │   └── ms@0.7.2
       ├─┬ newrelic@1.39.0
       │ │ │ └── sprintf-js@1.0.3
       │ └── xtend@4.0.1
       ├─┬ morgan@1.8.1
       │ └─┬ debug@2.6.1
       │ ├─┬ concat-stream@1.6.0
       │ │ ├── inherits@2.0.3
       │ │ ├─┬ readable-stream@2.2.9
       │ │ │ ├── buffer-shims@1.0.0
       │ │ │ ├── core-util-is@1.0.2
       │ │ │ ├── isarray@1.0.0
       │ │ │ ├── process-nextick-args@1.0.7
       │ │ │ ├── string_decoder@1.0.0
       │ │ │ └── util-deprecate@1.0.2
       │ │ └── typedarray@0.0.6
       │ ├─┬ https-proxy-agent@0.3.6
       │ │ ├── agent-base@1.0.2
       │ │ ├─┬ debug@2.6.6
       │ │ │ └── ms@0.7.3
       │ │ └── extend@3.0.1
       │ ├── json-stringify-safe@5.0.1
       │ ├─┬ readable-stream@1.1.14
       │ │ ├── core-util-is@1.0.2
       │ │ ├── inherits@2.0.3
       │ │ ├── isarray@0.0.1
       │ │ └── string_decoder@0.10.31
       │ └── semver@5.3.0
       ├─┬ passport@0.3.2
       │ ├── passport-strategy@1.0.0
       │ └── pause@0.0.1
       ├─┬ passport-predix-oauth@0.1.13
       │ ├─┬ https-proxy-agent@1.0.0
       │ │ └─┬ agent-base@2.0.1
       │ │   └── semver@5.0.3
       │ ├─┬ passport-oauth@0.1.15
       │ │ ├── oauth@0.9.15
       │ │ └── passport@0.1.18
       │ └── pkginfo@0.2.3
       ├─┬ predix-fast-token@1.1.1
       │ └─┬ node-cache@4.1.1
       │   └── clone@2.1.1
       ├── predix-uaa-client@1.2.1
       ├─┬ redis@2.7.1
       │ ├── double-ended-queue@2.1.0-0
       │ ├── redis-commands@1.3.1
       │ └── redis-parser@2.6.0
       ├─┬ request@2.81.0
       │ ├── aws-sign2@0.6.0
       │ ├── aws4@1.6.0
       │ ├── caseless@0.12.0
       │ ├─┬ combined-stream@1.0.5
       │ │ └── delayed-stream@1.0.0
       │ ├── extend@3.0.1
       │ ├── forever-agent@0.6.1
       │ ├─┬ form-data@2.1.4
       │ │ └── asynckit@0.4.0
       │ ├─┬ har-validator@4.2.1
       │ │ ├─┬ ajv@4.11.8
       │ │ │ ├── co@4.6.0
       │ ├─┬ hawk@3.1.3
       │ │ ├── hoek@2.16.3
       │ │ └── sntp@1.0.9
       │ ├─┬ http-signature@1.1.1
       │ │ │ ├── extsprintf@1.0.2
       │ │ │ └─┬ json-stable-stringify@1.0.1
       │ │ │   └── jsonify@0.0.0
       │ │ └── har-schema@1.0.5
       │ │ ├── boom@2.10.1
       │ │ ├── cryptiles@2.0.5
       │ │ ├── assert-plus@0.2.0
       │ │ ├─┬ jsprim@1.4.0
       │ │ │ ├── assert-plus@1.0.0
       │ │ │ ├── json-schema@0.2.3
       │ │ │ └── verror@1.3.6
       │ │ └─┬ sshpk@1.13.0
       │ │   ├── asn1@0.2.3
       │ │   ├── assert-plus@1.0.0
       │ │   ├── bcrypt-pbkdf@1.0.1
       │ │   ├─┬ dashdash@1.14.1
       │ │   │ └── assert-plus@1.0.0
       │ │   ├── ecc-jsbn@0.1.1
       │ │   ├─┬ getpass@0.1.7
       │ │   │ └── assert-plus@1.0.0
       │ │   ├── jodid25519@1.0.2
       │ │   ├── jsbn@0.1.1
       │ │   └── tweetnacl@0.14.5
       │ ├── is-typedarray@1.0.0
       │ ├── isstream@0.1.2
       │ ├── json-stringify-safe@5.0.1
       │ ├── mime-types@2.1.15
       │ ├── oauth-sign@0.8.2
       │ ├── performance-now@0.2.0
       │ ├── qs@6.4.0
       │ ├── safe-buffer@5.0.1
       │ ├── stringstream@0.0.5
       │ ├─┬ tough-cookie@2.3.2
       │ │ └── punycode@1.4.1
       │ └── tunnel-agent@0.6.0
       ├─┬ request-promise@4.2.0
       │ ├── bluebird@3.5.0
       │ ├── request-promise-core@1.1.1
       │ └── stealthy-require@1.1.0
       ├─┬ sanitize-html@1.14.1
       │ ├─┬ htmlparser2@3.9.2
       │ │ ├── domelementtype@1.3.0
       │ │ ├── domhandler@2.3.0
       │ │ ├─┬ domutils@1.6.0
       │ │ │ └─┬ dom-serializer@0.1.0
       │ │ │   └── domelementtype@1.1.3
       │ │ └── entities@1.1.1
       │ └── regexp-quote@0.0.0
       ├─┬ serve-static@1.12.2
       │ └─┬ send@0.15.2
       │   ├─┬ debug@2.6.4
       │   │ └── ms@0.7.3
       │   ├── etag@1.8.0
       │   ├── fresh@0.5.0
       │   ├─┬ http-errors@1.6.1
       │   │ └── setprototypeof@1.0.3
       │   └── ms@1.0.0
       ├─┬ url@0.11.0
       │ ├── punycode@1.3.2
       │ └── querystring@0.2.0
       └── uuid@3.0.1
-----> Caching build
error: Invalid character
.cacheDirectories | .[]?
                       ^
1 compile error
error: Invalid character
.cache_directories | .[]?
                        ^
1 compile error
       Clearing previous node cache
       Saving 3 cacheDirectories (default):
       - .npm
       - .cache/yarn (nothing to cache)
       - bower_components (nothing to cache)
-----> Build succeeded!

**Comment by sclevine** [comment]
```
-----> Build succeeded!
```
As @sesmith177 mentioned, it looks like these errors don't break `cf push`, and that they can be ignored until we address them in the next release. Does your app fail to start?

**Comment by idgm81** [comment]
Some issue here and my app fails to start

**Comment by sclevine** [comment]
The next release implements a workaround for the CC issue and (separately) should clear up any caching-related warnings. Please follow this story: https://www.pivotaltracker.com/story/show/145044293

**Comment by sturadnidge** [comment]
Hi all - I've been seeing this problem too, it doesn't actually cause anything to fail but doesn't look great to end users.

This app consistently gets the problem https://github.com/sturadnidge/cf-demo-node against node buildpack 1.5.33

**Comment by sclevine** [comment]
Fixed in the latest release: [1.5.34](https://github.com/cloudfoundry/nodejs-buildpack/releases/tag/v1.5.34).
